src/librustc/diagnostics.rs

```diff
@@ -1444,6 +1444,51 @@ lint name). Ensure the attribute is of this form:
 ```
 "##,
 
+E0453: r##"
+A lint check attribute was overruled by a `forbid` directive set as an
+attribute on an enclosing scope, or on the command line with the `-F` option.
+
+Example of erroneous code:
+
+```compile_fail
+#![forbid(non_snake_case)]
+
+#[allow(non_snake_case)]
+fn main() {
+    let MyNumber = 2; // error: allow(non_snake_case) overruled by outer
+                      // forbid(non_snake_case)
+}
+```
+
+The `forbid` lint setting makes code that fails the lint check result in a
```

Comment:
"Compilation warning" may not be the most _strictly_ accurate phrase—the lint check emits a compilation warning if it's set to `warn`, but it could default to `allow`, _&c._—but I agree that it seems worth glossing over such fine distinctions and preferring familiar phrases for the sake of pedagogy.